### PR TITLE
fix add appointments and implements add appointments recursively 

### LINKED ISSUE #39: As a clerk, I can add new appointments recursively for a patient

### PR DESCRIPTION
**Pull request by woon17**
close #39

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddAppCommand.java [hunk 1]
```diff
@@ -28,31 +33,73 @@
...
         requireNonNull(model);
+        if (eventList == null) {
+            addOneEvent(model);
+            model.updateFilteredEventList(toAdd.getPersonId());
+            return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
+
+        } else {
+            for (Event e : eventList) {
+                if (model.hasEvent(e)) {
+                    throw new CommandException(MESSAGE_DUPLICATE_EVENT);
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/logic/commands/AddAppCommand.java]: Could you clarify in the UG and DG, what would happen if only 1 of mutliple recurring events conflict with another event? Should it all fail? Or should only that single recurring event fail?

File: src/main/java/seedu/address/logic/parser/AddAppCommandParser.java [hunk 2]
```diff
@@ -58,9 +66,52 @@ public ReversibleActionPairCommand parse(String args) throws ParseException {
...
+            return new ReversibleActionPairCommand(new AddAppCommand(event),
+                    new CancelAppCommand(event));
+        }
+    }
+
+    private List<Event> getRecEvents(Appointment event, String recursiveString, int times) {
+        List<Event> eventList = new ArrayList<>();
+        Timing timing = event.getEventTiming();
+
+        if (recursiveString.equals("w")) {
```
Comment (SakuraBlossom) [inline src/main/java/seedu/address/logic/parser/AddAppCommandParser.java]: Could you add an option for daily?

Comment (woon17) [reply src/main/java/seedu/address/logic/parser/AddAppCommandParser.java]: okay

File: src/main/java/seedu/address/logic/parser/AddAppCommandParser.java [hunk 3]
```diff
@@ -58,9 +66,52 @@ public ReversibleActionPairCommand parse(String args) throws ParseException {
...
+                timing = Timing.getOneMonthLaterTiming(timing);
+            }
+        } else {
+            for (int i = 0; i < times; i++) {
+                eventList.add(new Appointment(event.getPersonId(), timing, new Status()));
+                timing = Timing.getOneYearLaterTiming(timing);
+
+            }
+        }
+        return eventList;
```
Comment (CarbonGrid) [inline src/main/java/seedu/address/logic/parser/AddAppCommandParser.java]: Suggests to simplify if else block.
```
private List<Event> getRecEvents(Appointment event, String recursiveString, int times) {
    List<Event> eventList = new ArrayList<>();
    Timing timing = event.getEventTiming();
    Function<Timing, Timing> func = null;
    
    switch(recursiveString) {
    case "w":
        func = Timing::getOneWeekLaterTiming;
        break;
    case "m":
        func = Timing::getOneMonthLaterTiming;
        break;
    default:
        func = Timing::getOneYearLaterTiming;
        break;
    }
    
    for (int i = 0; i < times; i++) {
        eventList.add(new Appointment(event.getPersonId(), timing, new Status()));
        timing = func.apply(timing);
    }
    
    return eventList;
}
```

Comment (woon17) [reply src/main/java/seedu/address/logic/parser/AddAppCommandParser.java]: okay

**Pull request merged**